### PR TITLE
Issue #42_Create API for getting playlist of station

### PR DESCRIPTION
**Pull request by Tran-Hoang-Lam**
Dear @dat-ng 
I finished issue 42 [https://github.com/mgm-interns/teamradio2-frontend/issues/42]

Please help me review this.
Thanks!

### CODE REVIEW

File: src/main/java/com/mgmtp/radio/controller/v1/SongController.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Request processed successfully"),
+            @ApiResponse(code = 200, message = "No message then there is error, connection close")
+    })
+    @GetMapping("/sse/listSong")
+    public Flux<ServerSentEvent<RadioSuccessResponse<List<SongDTO>>>> getListSong(@RequestParam("stationId") String stationId) {
+        return Flux.interval(Duration.ofSeconds(1))
+                .map(tick -> Tuples.of(tick, songService.getListSongBy(stationId)))
+                .map(tuple2 -> {
```
Comment (dat-ng) [inline src/main/java/com/mgmtp/radio/controller/v1/SongController.java]: change name

File: src/main/java/com/mgmtp/radio/controller/v1/SongController.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                    return ServerSentEvent.<RadioSuccessResponse<List<SongDTO>>>builder()
+                            .event("fetch")
+                            .id(Long.toString(tuple2.getT1()))
+                            .data(
+                                    new RadioSuccessResponse<>(
+                                            tuple2
+                                                    .getT2()
```
Comment (dat-ng) [inline src/main/java/com/mgmtp/radio/controller/v1/SongController.java]: change name

File: src/main/java/com/mgmtp/radio/service/station/SongService.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.mgmtp.radio.service.station;
+
+import com.mgmtp.radio.dto.station.SongDTO;
+import reactor.core.publisher.Flux;
+
+public interface SongService {
+    Flux<SongDTO> getListSongBy(String stationId);
```
Comment (dat-ng) [inline src/main/java/com/mgmtp/radio/service/station/SongService.java]: ByStationId, getListSong(String stationId)

File: src/main/java/com/mgmtp/radio/service/station/SongServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        this.songMapper = songMapper;
+        this.userMapper = userMapper;
+    }
+
+    @Override
+    public Flux<SongDTO> getListSongBy(String stationId) {
+        return stationRepository.findByIdAndDeletedFalse(stationId).flatMapMany(station -> {
+            List<String> playList = station.getPlaylist();
+
+            return songRepository.findByIdIn(playList).map(song -> {
```
Comment (dat-ng) [inline src/main/java/com/mgmtp/radio/service/station/SongServiceImpl.java]: should separate new private method

**Comment by Tran-Hoang-Lam** [comment]
Dear @dat-ng 
I fixed your comment, please review

Thank you!

**Pull request merged**